packages/taiko-client/driver/preconf_blocks/server.go

```diff
@@ -511,8 +995,10 @@ func (s *PreconfBlockAPIServer) UpdateLookahead(l *Lookahead) {
 	s.lookahead = l
 }
 
-// checkLookaheadHandover returns nil if feeRecipient is allowed to build at slot globalSlot (absolute L1 slot).
-func (s *PreconfBlockAPIServer) checkLookaheadHandover(feeRecipient common.Address, globalSlot uint64) error {
+// CheckLookaheadHandover returns nil if feeRecipient is allowed to build at slot globalSlot (absolute L1 slot).
+// and checks the  handover window to see if we need to request the end of sequencing
+// block.
+func (s *PreconfBlockAPIServer) CheckLookaheadHandover(feeRecipient common.Address, globalSlot uint64) error {
 	s.lookaheadMutex.Lock()
 	la := s.lookahead
```

Comment:
It's just a copy of a pointer? shall we change the `s.lookaheadMutex.Unlock()` to `defer s.lookaheadMutex.Unlock()`?

Reply:
done